lib/fetch/body.js

```diff
@@ -19,33 +19,65 @@ const { parseMIMEType, serializeAMimeType } = require('./dataURL')
 /** @type {globalThis['ReadableStream']} */
 let ReadableStream
 
-async function * blobGen (blob) {
-  yield * blob.stream()
-}
-
 // https://fetch.spec.whatwg.org/#concept-bodyinit-extract
 function extractBody (object, keepalive = false) {
   if (!ReadableStream) {
     ReadableStream = require('stream/web').ReadableStream
   }
 
-  // 1. Let stream be object if object is a ReadableStream object.
-  // Otherwise, let stream be a new ReadableStream, and set up stream.
+  // 1. Let stream be null.
   let stream = null
 
-  // 2. Let action be null.
+  // 2. If object is a ReadableStream object, then set stream to object.
+  if (object instanceof ReadableStream) {
+    stream = object
+  } else if (isBlobLike(object)) {
+    // 3. Otherwise, if object is a Blob object, set stream to the
+    //    result of running object’s get stream.
+    stream = object.stream()
+  } else {
+    // 4. Otherwise, set stream to a new ReadableStream object, and set
+    //    up stream.
+    stream = new ReadableStream({
+      async pull (controller) {
+        controller.enqueue(
+          typeof source === 'string' ? new TextEncoder().encode(source) : source
+        )
+        queueMicrotask(() => {
```

Comment:
Yes, this is a known issue with node readable streams. I think we have the same queueMicrotask hack somewhere else.

Reply:
Maybe we could move this into a `closeReadableStream` helper function and re-use? Then we can link additional information why that is needed in one place.

Reply:
Yeah I see it somewhere else, I actually copied it from there lol

Do you know why it happens?

Reply:
I don't remember. Maybe @jasnell has an idea. I believe there is some issue over at the node repo.

Reply:
added the helper